more_itertools/more.py

```diff
@@ -1214,3 +1215,65 @@ def groupby_transform(iterable, keyfunc=None, valuefunc=None):
     """
     valuefunc = (lambda x: x) if valuefunc is None else valuefunc
     return ((k, map(valuefunc, g)) for k, g in groupby(iterable, keyfunc))
+
+
+def arithmetic_sequence(*args):
```

Comment:
Why not use the word "range" in the name, both for brevity and to suggest similarity with the builtin of the same name? math_range, arithmetic_range, numeric_range